src/Storages/ObjectStorage/StorageObjectStorageSource.cpp

```diff
@@ -121,11 +123,14 @@ std::shared_ptr<StorageObjectStorageSource::IIterator> StorageObjectStorageSourc
     std::unique_ptr<IIterator> iterator;
     if (configuration->isPathWithGlobs())
     {
+        bool throw_on_zero_files_match = settings.throw_on_zero_files_match;
+        if (override_settings_for_hive_partitioning)
+            throw_on_zero_files_match = false;
```

Comment:
I don't like this way of overriding `throw_on_zero_files_match`. Propogating `override_settings_for_hive_partitioning` bool flag to `createFileIterator` looks a bit hacky to me because `createFileIterator` should know nothing about hive partitioning.
For me it would be better to do on of these 2 variants:
- Temporary override `throw_on_zero_files_match` setting in configurating in `StorageObjectStorage::getPathSample`.
- Propogate flag `throw_on_zero_files_match` to `createFileIterator` and set it to false in `getPathSample`.

I would prefer the first variant, but if you don't like it, you can do the second one